modules/core/test/test_arithm.cpp

```diff
@@ -1912,3 +1912,13 @@ TEST(Subtract, scalarc4_matc4)
 
     ASSERT_EQ(0, cv::norm(cv::Mat(5, 5, CV_8UC4, cv::Scalar::all(250)), destImage, cv::NORM_INF));
 }
+
+TEST(Compare, empty)
+{
+    cv::Mat temp, dst1, dst2;
+    cv::compare(temp, temp, dst1, cv::CMP_EQ);
+    dst2 = temp > 5;
```

Comment:
@alalek, this is rather philosophical question. I believe that some cases can be "eaten". In this particular case the error will likely be signalled later when user will try to analyze the comparison result. And it will likely be signalled as exception, not as a program crash.